backend/src/email/middlewares/email-template.middleware.ts

```diff
@@ -246,6 +248,44 @@ export const InitEmailTemplateMiddleware = (
     }
   }
 
+  const selectPhonebookListHandler = async (
+    req: Request,
+    res: Response
+  ): Promise<Response | void> => {
+    const { campaignId } = req.params
+
+    const { list_id: listId } = req.body
+
+    const list = await PhonebookService.getPhonebookListById({
+      listId,
+    })
+    if (!list) throw new Error('Error: List not found')
+
+    const { s3Key, etag, filename } = list
+
+    // check if template exists
+    const template = await EmailTemplateService.getFilledTemplate(+campaignId)
+    if (template === null) {
+      throw new Error(
```

Comment:
shouldnt this step be done before invoking phonebook as it unnecessarily creates a s3 file ?